common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java

```diff
@@ -0,0 +1,188 @@
+                        entry.setValue(decryptedText);
+                    }
+                },
+                result,
+                params,
+                keysMarkedForRemoval,
+                skipKeys,
+                shouldAbort
+        );
+
+        // Clear any keys marked for removal...
+        clearEntriesMarkedForRemoval(fileManager, cacheEntries, keysMarkedForRemoval);
+
+        if (shouldAbort.get()) {
+            Logger.info(TAG + methodName, "Aborting after decrypt.");
+            return result;
+        }
+
+        // Reencrypt everything
+        applyCacheMutation(
+                cacheEntries,
+                new Callable<Map.Entry<String, String>>() {
+                    @Override
+                    public void call(final Map.Entry<String, String> entry) throws Exception {
+                        final String reencryptedText = encrypter.encrypt(entry.getValue());
+                        entry.setValue(reencryptedText);
+                    }
+                },
+                result,
+                params,
+                keysMarkedForRemoval,
+                skipKeys,
+                shouldAbort
+        );
+
+        // Clear any keys marked for removal...
+        clearEntriesMarkedForRemoval(fileManager, cacheEntries, keysMarkedForRemoval);
+
+        if (shouldAbort.get()) {
+            Logger.info(TAG + methodName, "Aborting after reencrypt.");
+            return result;
+        }
+
+        // Write the newly encrypted records...
+        Logger.info(TAG + methodName, "Writing reencrypted cache entries.");
+        for (final Map.Entry<String, String> cacheEntry : cacheEntries.entrySet()) {
+            fileManager.putString(cacheEntry.getKey(), cacheEntry.getValue());
+        }
+
+        return result;
+    }
+
+    private interface Callable<T> {
+        void call(T t) throws Exception;
+    }
+
+    private void applyCacheMutation(@NonNull final Map<String, String> cacheEntries,
+                                    @NonNull final Callable<Map.Entry<String, String>> callable,
+                                    @NonNull final MigrationOperationResult inputResult,
+                                    @NonNull final ReencryptionParams params,
+                                    @NonNull final Set<String> keysMarkedForRemoval,
+                                    @NonNull final Set<String> skipKeys,
+                                    @NonNull AtomicBoolean shouldAbort) {
+        final String methodName = ":applyCacheMutation";
+        for (final Map.Entry<String, String> cacheEntry : cacheEntries.entrySet()) {
+            try {
+                if (skipKeys.contains(cacheEntry.getKey())) {
+                    Logger.warn(TAG + methodName, "Skipping entry.");
+                    continue;
+                }
+                callable.call(cacheEntry);
+            } catch (final Exception e) {
+                Logger.error(TAG + methodName, "Error during mutation", e);
```

Comment:
Do we need to log the key that caused the error? I guess it could be pii/oii?

Reply:
It will be both PII and OII - I can look at logging it via the PII/OII safe methods

Reply:
To keep this work easy to `cherry-pick` for @AdamBJohnsonx - I have applied your feedback in a single amended commit via `32fdc9cdf4a9357bd40cca2e0e08f6daf1d50157`